Std/Data/MLList/Meta.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Std.Data.MLList.IO
+import Std.Data.List.Basic
+
+/-!
+# Parallelization in Lean's tactic monads.
+
+`MetaM.runGreedily` will run a list of `MetaM α` in parallel, returning
+* a cancellation hook and
+* an `MLList MetaM α` that returns the results (and the relevant `MetaM` state)
+  in the order that they finish.
+
+After calling the cancellation hook the behaviour of the monadic lazy list should not be relied on.
+It may continue returning values, or fail.
+Recommended usage is to take a prefix of the list
+(e.g. with `MLList.takeUpToFirst` followed by `MLList.force`, or `MLList.takeAsList`)
+and then call the cancellation hook to request cancellation of later unwanted tasks.
+
+(Similarly also for `CoreM`, `TermElabM`, and `TacticM`.)
+
+## Implementation notes:
+Calling `IO.cancel` on `t.map f` does not cancel `t`,
+so we have to be careful throughout this file
+to construct cancellation hooks connected to the underlying task,
+rather than the various maps of it that we construct to pass state.
+
+Thomas Murrills has a suggestion to significantly refactor this code,
```

Comment:
```suggestion
In several places below we use `prio := .max` to set the priority of tasks higher. This is only used on `map` tasks which run after the main workload has finished. This helps avoid the situation where the main workload has finished, but the value is not returned to the consumer because other long-running tasks are scheduled before the `map`.

Thomas Murrills has a suggestion to significantly refactor this code,
```